include directory.hpp explicitly; required with boost 1.85+
backport of commit 264c2fe

Index: src/stream/slice.cpp
--- src/stream/slice.cpp.orig
+++ src/stream/slice.cpp
@@ -27,6 +27,7 @@
 #include <boost/cstdint.hpp>
 #include <boost/algorithm/string/predicate.hpp>
 #include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/directory.hpp>
 #include <boost/range/size.hpp>
 
 #include "util/console.hpp"
